src/MapboxStyleParser.ts

```diff
@@ -699,11 +715,21 @@ export class MapboxStyleParser implements StyleParser<Omit<MbStyle, 'sources'>>
           scaleDenominator,
           symbolizers
         };
-        gsRules?.push(rule);
+        gsRules.push(rule);
+        if (symbolizers.length > 1) {
+          splitSymbolizers.push({
+            rule: gsRules.length - 1,
+            symbolizers: symbolizers.map((s, sIdx) => sIdx)
+          });
+        }
       });
     }
 
-    return gsRules || [];
+    if (splitSymbolizers.length) {
+      mapboxRef.splitSymbolizers = splitSymbolizers;
```

Comment:
I think we shouldn't modify the mapboxRef inside this function directly but return a modified clone together with the rules. This will avoid sideffects.